gusto/state.py

```diff
@@ -210,12 +316,26 @@ def setup_dump(self, pickup=False):
             field = Function(functionspaceimpl.WithGeometry(f.function_space(), mesh_ll), val=f.topological, name=name+'_ll')
             self.to_dump_latlon.append(field)
 
-    def dump(self, t=0, diagnostic_everydump=False, pickup=False):
+        # we create new netcdf files to write to, unless pickup=True, in
+        # which case we just need the filenames
+        diagnostics_filename = self.dumpdir+"/diagnostics.nc"
+
+        pointdata_filename = self.dumpdir+"/point_data.nc"
```

Comment:
The directory name is provided by the user and lives in ```self.output```. We automatically create this directory in a ```results``` subdirectory (see line 288) so the difference is that ```self.dumpdir``` has ```results/``` prepended to it - just saves the user typing it repeatedly and keeps the directory tidy. Happy to change if it doesn't make sense to other people though!